testing/test-setup/src/lib/extend/markdown-table.matcher.ts

```diff
@@ -0,0 +1,44 @@
+import type { SyncExpectationResult } from '@vitest/expect';
+import { expect } from 'vitest';
+
+export type CustomMarkdownTableMatchers = {
+  toContainMarkdownTableRow: (cells: string[]) => void;
+};
+
+expect.extend({
+  toContainMarkdownTableRow: assertMarkdownTableRow,
+});
+
+function assertMarkdownTableRow(
+  actual: string,
+  expected: string[],
+): SyncExpectationResult {
+  const rows = actual
+    .split('\n')
+    .map(line => line.trim())
+    .filter(line => line.startsWith('|') && line.endsWith('|'))
+    .map(line =>
+      line
+        .split('|')
```

Comment:
There's an edge case where this simple split by `|` might get tripped up. A Markdown cell could contain an escaped `\|`, which shouldn't be interpreted as a column separator.

E.g. this markdown:

```md
| Package  | Versions             |
| :------- | :------------------- |
| `eslint` | `^8.0.0 \|\| ^9.0.0` |
```

renders 2 columns and the escaped bars turn into cell content:

| Package  | Versions             |
| :------- | :------------------- |
| `eslint` | `^8.0.0 \|\| ^9.0.0` |

To handle these cases properly, you can use a regex with negative lookbehind as the separator:

```suggestion
        .split(/(?<!\\)\|/)
```